app/controllers/state_file/questions/az_public_school_contributions_controller.rb

```diff
@@ -0,0 +1,90 @@
+module StateFile
+  module Questions
+    class AzPublicSchoolContributionsController < QuestionsController
+      include ReturnToReviewConcern
+      include DateHelper
+
+      def self.navigation_actions
+        [:index, :new]
+      end
+
+      def index
+        @az322_contributions = current_intake.az322_contributions
+        unless @az322_contributions.present?
+          redirect_to action: :new
+        end
+      end
+
+      def new
+        @az322_contribution = current_intake.az322_contributions.build
+        @filing_year = Rails.configuration.statefile_current_tax_year
+      end
+
+      def create
+        @az322_contribution = current_intake.az322_contributions.build(az322_contribution_params)
+        @filing_year = Rails.configuration.statefile_current_tax_year
+        @az322_contributions = current_intake.az322_contributions
+        if @az322_contribution.made_contribution_no?
+          return redirect_to next_path
+        end
+
+        if @az322_contribution.valid?
+          @az322_contribution.save
+          redirect_to action: :index, return_to_review: params[:return_to_review]
+        else
+          render :new
+        end
+      end
+
+      def edit
+        @az322_contribution = current_intake.az322_contributions.find(params[:id])
+        @filing_year = Rails.configuration.statefile_current_tax_year
+      end
+
+      def update
+        @az322_contribution = current_intake.az322_contributions.find(params[:id])
+        @filing_year = Rails.configuration.statefile_current_tax_year
```

Comment:
since this is used for several actions i wonder if we can just add it to the top instead in something like `before_action :load_filing_year, only: [:new, :create, :edit, :update]` and def load_filing_year; @filing_year = Rails.configuration.statefile_current_tax_year; end;`